crates/oxc_linter/src/rules/eslint/no_lonely_if.rs

```diff
@@ -0,0 +1,296 @@
+use oxc_ast::AstKind;
+use oxc_ast::ast::{Statement, Statement::BlockStatement};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+fn no_lonely_if_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Unexpected `if` as the only statement in an `else` block")
+        .with_help("Consider using `else if` instead.")
+        .with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoLonelyIf;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallow `if` statements as the only statement in `else` blocks
+    ///
+    /// ### Why is this bad?
+    ///
+    /// When an `if` statement is the only statement in an `else` block, it is often clearer to use
+    /// an `else if` instead.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// if (condition) {
+    ///   // ...
+    /// } else {
+    ///   if (anotherCondition) {
+    ///     // ...
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// ```js
+    /// if (condition) {
+    ///   // ...
+    /// } else {
+    ///   if (anotherCondition) {
+    ///     // ...
+    ///   } else {
+    ///       // ...
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// if (condition) {
+    ///   // ...
+    /// } else if (anotherCondition) {
+    ///   // ...
+    /// }
+    /// ```
+    ///
+    /// ```js
+    /// if (condition) {
+    ///   // ...
+    /// } else if (anotherCondition) {
+    ///   // ...
+    /// } else {
+    ///   // ...
+    /// }
+    /// ```
+    ///
+    /// ```js
+    /// if (condition) {
+    ///   // ...
+    /// } else {
+    ///   if (anotherCondition) {
+    ///     // ...
+    ///   }
+    ///   doSomething();
+    /// }
+    /// ```
+    NoLonelyIf,
+    eslint,
+    pedantic,
+    pending
+);
+
+impl Rule for NoLonelyIf {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::IfStatement(if_stmt) = node.kind() else {
+            return;
+        };
+
+        let Some(ref alternate) = if_stmt.alternate else {
+            return;
+        };
+
+        let BlockStatement(b) = alternate else {
+            return;
+        };
+
+        let Some(only_stmt) = b.body.first() else {
+            return;
+        };
+
+        if let Some(AstKind::IfStatement(_)) = ctx.nodes().parent_kind(node.id()) {
+            return;
+        };
+
+        if b.body.len() != 1 {
+            return;
+        };
+
+        if let Statement::BlockStatement(inner_block) = only_stmt {
+            if inner_block.body.len() == 1 {
+                if let Some(Statement::IfStatement(lonely_if)) = inner_block.body.first() {
+                    ctx.diagnostic(no_lonely_if_diagnostic(Span::new(
+                        lonely_if.span.start,
+                        lonely_if.span.start + 2,
+                    )));
```

Comment:
can use `Span::sized` to make this shorter:
```suggestion
                    ctx.diagnostic(no_lonely_if_diagnostic(Span::sized(
                        lonely_if.span.start,
                        2,
                    )));
```